app/jobs/change_pickup_library_job.rb

```diff
@@ -16,19 +16,30 @@ def perform(hold_key:, session_token:, pickup_library:)
     when 200
       human_readable_location = Hold::PICKUP_LOCATION_ACTUAL[pickup_library.to_sym]
       Redis.current.set("pickup_library_#{hold_key}", {
-        hold_id: hold_key,
+        id: hold_key,
         result: :success,
-        new_value: human_readable_location,
-        new_value_id: pickup_library
+        response: {
+          new_value: human_readable_location,
+          new_value_id: pickup_library
+        }
       }.to_json)
     else
-      error_message_raw = JSON.parse response.body
-      error_message = error_message_raw&.dig('messageList')&.first&.dig('message') || 'Something went wrong'
-      Sidekiq.logger.error("pickup_library_#{hold_key}: #{error_message}")
+      processed_error = SirsiResponse::Error.new(error_message_raw: JSON.parse(response.body),
+                                                 symphony_client: symphony_client,
+                                                 key: hold_key,
+                                                 session_token: session_token,
+                                                 bib_type: :hold)
+
+      Sidekiq.logger.error("pickup_library_#{hold_key}: #{processed_error.log}")
+
       Redis.current.set("pickup_library_#{hold_key}", {
-        hold_id: hold_key,
+        id: hold_key,
         result: :failure,
-        response: error_message
+        response: {
+          new_value: 'Error',
+          new_value_id: 'Error'
```

Comment:
`new_value_id: 'Error'` is this supposed to be `Error`?

Reply:
Nope. This is the value that gets processed by javascript as the new value in the table of holds. It'll say "Error" instead of whatever the new value should have been. Though I can't generate hold errors anymore infuriatingly.

Reply:
this is also related to this https://github.com/psu-libraries/myaccount/pull/182#discussion_r449353149, right?

Reply:
No, that comment is in regards to what gets sent back to the client as `display_error` (what shows up in the toast)

Reply:
Oops sorry added the wrong the link, meant to link to this convo https://github.com/psu-libraries/myaccount/pull/182#discussion_r449353998

Reply:
So, basically if we decide to keep/send the current values, these would not have the value `Error`, is what I was trying to ask :)

Reply:
Yeah, this part would need changing/rethinking